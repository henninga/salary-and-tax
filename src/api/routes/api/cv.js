import express from 'express';
import getCV from '../../cvpartner/getCV';

const router = express.Router();

router.get('/:userId/:cvId', (req, res) => {
    getCV(req.params)
        .then(cv => {
            res.status(200).json(cv);
        })
        .catch(({ error: { status, error } }) => {
            res.status(status).json({ error });
        });
});

module.exports = router;
