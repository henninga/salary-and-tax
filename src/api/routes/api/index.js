import express from 'express';
import cvRouter from './cv';
import usersRouter from './users';

const router = express.Router();

router.use('/users', usersRouter);
router.use('/cv', cvRouter);

module.exports = router;
