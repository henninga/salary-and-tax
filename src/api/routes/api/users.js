import express from 'express';
import getUsers from '../../cvpartner/getUsers';

const router = express.Router();

router.get('/', (req, res) => {
    getUsers()
        .then(users => {
            res.status(200).json(users);
        })
        .catch(({ error: { status, error } }) => {
            res.status(status).json({ error });
        });
});

module.exports = router;
