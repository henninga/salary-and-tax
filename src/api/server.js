#!/usr/bin/env node

import debug from 'debug';
import http from 'http';
import app from './app';

const logger = debug('technologic.as:server');

const normalizePort = (val) => {
    const port = parseInt(val, 10);

    if (isNaN(port)) { return val; }
    if (port >= 0) { return port; }

    return false;
};

const onError = (error) => {
    if (error.syscall !== 'listen') { throw error; }

    const bind = typeof port === 'string' ? `Pipe ${port}` : `Port ${port}`;

    switch (error.code) {
        case 'EACCES':
            logger(`ERROR: ${bind} requires elevated privileges`);
            process.exit(1);
            break;
        case 'EADDRINUSE':
            logger(`ERROR: ${bind} is already in use`);
            process.exit(1);
            break;
        default:
            throw error;
    }
};

const onListening = () => {
    const address = server.address();
    const bind = typeof address === 'string' ? `pipe ${address}` : `port ${address.port}`;
    logger(`Listening on ${bind}`);
};

const port = normalizePort(process.env.PORT || '3000');
app.set('port', port);

const server = http.createServer(app);

server.listen(port);
server.on('error', onError);
server.on('listening', onListening);
