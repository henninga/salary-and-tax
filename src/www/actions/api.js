import fetch from 'cross-fetch';


export const RECEIVE_USERS = 'RECEIVE_USERS';
export const REQUEST_USERS = 'REQUEST_USERS';
export const USER_IS_SELECTED = 'USER_IS_SELECTED';

export const REQUEST_CV = 'REQUEST_CV';
export const RECEIVE_CV = 'RECEIVE_CV';

export const FORM_CHANGE = 'FORM_CHANGE';

export const requestUsers = () => ({
    type: REQUEST_USERS,
});

export const receiveUsers = (list) => ({
    type: RECEIVE_USERS,
    list,
});

export const requestCv = (userId, cvId, name) => ({
    type: REQUEST_CV,
    userId,
    cvId,
    name,
});

export const receiveCv = (cv) => ({
    type: RECEIVE_CV,
    cv,
});

export const formChange = (values) => ({
  type: FORM_CHANGE,
  ...values,
});

export const fetchUsers = () => {
    return dispatch => {
        dispatch(requestUsers());
        return fetch('http://localhost:3000/api/users')
            .then(response => response.json())
            .then(list => dispatch(receiveUsers(list)))
            .then((users) => {
                if (users.list.length === 1) { dispatch(selectUser(users.list[0])) }
                return users;
            });
    }
};

export const selectUser = ({userId, cvId, name}) => {
    return dispatch => {
        dispatch(requestCv(userId, cvId, name));
        return fetch(`http://localhost:3000/api/cv/${userId}/${cvId}`)
            .then(response => response.json())
            .then(json => dispatch(receiveCv(json)))
    }
};

export const inputHasChanged = (values) => {
  return dispatch => {
    dispatch(formChange(values));
  }
};
