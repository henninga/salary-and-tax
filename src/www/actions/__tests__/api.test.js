import { requestCv, requestUsers } from '../api';

describe('requestUsers', () => {
    it('should return action matching snapshot', function () {
        expect(requestUsers()).toMatchSnapshot();
    });
});

describe('requestCv', () => {
    it('should return action matching snapshot', function () {
        expect(requestCv('idUser', 'idCv')).toMatchSnapshot();
    });
});
