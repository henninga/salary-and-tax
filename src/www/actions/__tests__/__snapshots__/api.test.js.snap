// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requestCv should return action matching snapshot 1`] = `
Object {
  "cvId": "idCv",
  "name": undefined,
  "type": "REQUEST_CV",
  "userId": "idUser",
}
`;

exports[`requestUsers should return action matching snapshot 1`] = `
Object {
  "type": "REQUEST_USERS",
}
`;
